server/WFCORE-6750-unstable-api-annotation-scanner.adoc

```diff
@@ -0,0 +1,243 @@
+
+== Test Plan
+
+Tests will be added in both WildFLy and WildFly Core
+
+WildFly Core:
+
+* The WildFly core feature pack is only used internally for testing, so we will not scan for annotations indicating unstable API with the plugin here. Also, it should be noted that the  usage of such annotations in third party components is outside of our control, and likely to change (e.g. an annotated member might 'mature' to no longer have the annotation, or the member might be removed in a future release). Meanwhile, the tests for the scanning mechanism should be in WildFly Core since that is where the deployment unit processors live.
+** Thus, a test feature pack will be created offering an API (including its own annotations for indicating unstable API) and a subsystem providing this API on the classpath.
+** The feature pack will be indexed and the indices will be added to the `{indexModule}` module.
+** The test will check that
+*** the index file is added to the `${indexModule}` module, and to the list in the module's `index.txt`, as outlined in the requirements
+*** when deploying an application using annotated API elements from the test feature pack API, usage of those is reported in line with the configuration in the `subsystem=core-management/service=unstable-api-annotations` resource (as outlined in the requirements). We will test both with
+**** `level=log` - and inspect that the log messages pick up all the usages of annotated API elements.
+**** `level=error` - and inspect that the application fails to deploy, and that the error message picks up all the usages of annotated API elements.
+** The test will run at the `preview` stability level
+
+WildFly:
+
+* Since we have done the main testing of the mechanism in WildFly Core, we will not test that here
+* A test will be added to `testsuite/integration/basic` which runs in an execution provisioning both `wildfly-ee-feature-pack` and `wildfly-galleon-pack`. It will test
+** that the `{indexModule}` module, and the list in the module's `index.txt`, contain the indices from both feature packs.
+** A number of differently packaged deployments will be tested, to make sure that each class only gets scanned once, in order to test that the annotated class/member usage follows the WildFly classloading rules. The scanner outputs the class count when an undocumented system property is set to `true`.
+* The above test will run at the `preview` stability level
+
+== Community Documentation
+////
+Generally a feature should have documentation as part of the PR to wildfly master, or as a follow up PR if the feature is in wildfly-core. In some cases though the documentation belongs more in a component, or does not need any documentation. Indicate which of these will happen.
+////
+
+Community documentations will be added as part of the PR to WildFly full. This PR will also contain the parts of the tests which live in WildFly, as mentioned in the previous section
```

Comment:
We need to beef this bit up a lot. I think not implementing everything by the 33 feature freeze is ok, but we should think about things like user journeys.

Reply:
Actually I think what you did is ok for Preview. A blog would be good though, if not now then as part of moving to Community.